packages/big-design/src/components/Worksheet/hooks/useEditableCell/spec.tsx

```diff
@@ -0,0 +1,166 @@
+    const { result } = renderHook(() => useEditableCell(cell));
+
+    act(() => result.current.handleDoubleClick());
+
+    expect(result.current.isEditing).toBe(true);
+  });
+
+  test('blur', () => {
+    const { result } = renderHook(() => useEditableCell(cell));
+
+    act(() => {
+      result.current.handleDoubleClick();
+      result.current.handleBlur();
+    });
+
+    expect(result.current.isEditing).toBe(false);
+  });
+
+  test('enter', () => {
+    const { result } = renderHook(() => useEditableCell(cell));
+
+    act(() => {
+      result.current.handleDoubleClick();
+      result.current.handleKeyDown(
+        ({
+          key: 'Enter',
+          preventDefault,
+        } as unknown) as React.KeyboardEvent<HTMLInputElement>,
+        'New Value',
+      );
+    });
+
+    expect(preventDefault).toHaveBeenCalledTimes(1);
+    expect(mockUpdateItems).toHaveBeenCalledWith([cell], ['New Value']);
+    expect(result.current.isEditing).toBe(false);
+  });
+
+  test('enter only updates if it has new values', () => {
+    const { result } = renderHook(() => useEditableCell(cell));
+
+    act(() => {
+      result.current.handleDoubleClick();
+      result.current.handleKeyDown(
+        ({
+          key: 'Enter',
+          preventDefault,
+        } as unknown) as React.KeyboardEvent<HTMLInputElement>,
+        'Text',
+      );
+    });
+
+    expect(mockUpdateItems).not.toHaveBeenCalled();
+  });
+
+  test('tab', () => {
+    const { result } = renderHook(() => useEditableCell(cell));
+
+    act(() => {
+      result.current.handleKeyDown(
+        ({
+          key: 'Tab',
+          preventDefault,
+        } as unknown) as React.KeyboardEvent<HTMLInputElement>,
+        'New Value',
+      );
+    });
+
+    expect(mockNavigate).toHaveBeenCalledWith({ columnIndex: 1, rowIndex: 0 });
+
+    act(() => {
+      result.current.handleKeyDown(
+        ({
+          key: 'Tab',
+          preventDefault,
+          shiftKey: true,
+        } as unknown) as React.KeyboardEvent<HTMLInputElement>,
+        'New Value',
+      );
+    });
+
+    expect(mockNavigate).toHaveBeenCalledWith({ columnIndex: -1, rowIndex: 0 });
+
+    expect(result.current.isEditing).toBe(false);
+  });
+
+  test('modal and select do not trigger a navigation', () => {
+    const { result } = renderHook(() => useEditableCell({ ...cell, type: 'select' }));
+
+    act(() => {
+      result.current.handleKeyDown(
+        ({
+          key: 'Enter',
+          preventDefault,
+        } as unknown) as React.KeyboardEvent<HTMLInputElement>,
+        'New Value',
+      );
+    });
+
+    expect(mockNavigate).not.toHaveBeenCalled();
+  });
+
+  test('escape', () => {
+    const { result } = renderHook(() => useEditableCell(cell));
+
+    act(() => {
+      result.current.handleDoubleClick();
+      result.current.handleKeyDown(
+        ({
+          key: 'Escape',
+          preventDefault,
+        } as unknown) as React.KeyboardEvent<HTMLInputElement>,
```

Comment:
🍹 hmm, i wonder if we should invest in getting rid of all of these `as` type casts, but not sure how strict yall are here in big-design

Reply:
I wouldn't know how to use this without ts breaking. Maybe override the rule? Different fix, same problem?